share/getters/shrex.go

```diff
@@ -79,6 +136,7 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 				"hash", root.String(),
 				"err", getErr,
 				"finished (s)", time.Since(start))
+			sg.metrics.recordEDSAttempt(ctx, attempt, false)
```

Comment:
if you use `ctx` here metric will not be submitted, since ctx already timeout out.